release_notes/ocp_3_10_release_notes.adoc

```diff
@@ -54,33 +54,524 @@ documentation.
+
+CPU Manager is useful for workloads that have some of these attributes:
+
+* Require as much CPU time as possible.
+* Are sensitive to processor cache misses.
+* Are low-latency network applications.
+* Coordinate with other processes and benefit from sharing a single processor
+cache.
+
+See
+xref:../scaling_performance/using_cpu_manager.adoc#scaling-performance-using-cpu-manager[Using
+CPU Manager] for more information.
+
+[[ocp-310-device-manager]]
+==== Device Manager
+
+Device Manager is now moved out of Technology Preview and generally available in
+{product-title} 3.10.
+
+Some users want to set resource limits for hardware devices within their pod
+definition and have the scheduler find the node in the cluster with those
+resources.  While at the same time, Kubernetes needed a way for hardware
+vendors to advertise their resources to the kubelet without forcing them to
+change core code within Kubernetes
+
+The kubelet now houses a device manager that is extensible through plug-ins. You
+load the driver support at the node level. Then, you or the vendor writes a
+plug-in that listens for requests to stop/start/attach/assign the requested
+hardware resources seen by the drivers. This plug-in is deployed to all the
+nodes via a daemonSet.
+
+See xref:../dev_guide/device_manager.adoc#using-device-manager[Using Device
+Manager] for more information.
+
+[[ocp-310-hugepages]]
+==== Huge Pages
+
+Huge pages are now moved out of Technology Preview and generally available in
+{product-title} 3.10.
+
+Memory is managed in blocks known as pages. On most systems, a page is 4Ki. 1Mi
+of memory is equal to 256 pages; 1Gi of memory is 256,000 pages, and so on. CPUs
+have a built-in memory management unit that manages a list of these pages in
+hardware. The Translation Lookaside Buffer (TLB) is a small hardware cache of
+virtual-to-physical page mappings. If the virtual address passed in a hardware
+instruction can be found in the TLB, the mapping can be determined quickly. If
+not, a TLB miss occurs, and the system falls back to slower, software-based
+address translation, resulting in performance issues. Since the size of the
+TLB is fixed, the only way to reduce the chance of a TLB miss is to increase the
+page size.
+
+A huge page is a memory page that is larger than 4Ki. On x86_64 architectures,
+there are two common huge page sizes: 2Mi and 1Gi. Sizes vary on other
+architectures. In order to use huge pages, code must be written so that
+applications are aware of them. Transparent Huge Pages (THP) attempt to automate
+the management of huge pages without application knowledge, but they have
+limitations. In particular, they are limited to 2Mi page sizes. THP can lead to
+performance degradation on nodes with high memory utilization or fragmentation
+due to defragmenting efforts of THP, which can lock memory pages. For this
+reason, some applications may be designed to (or recommend) usage of
```

Comment:
Does this sentence `designed to (or recommend) usage of pre-allocated huge pages` have syntax error?